File: surplus-hub-api-v3/app/models/event.py
from sqlalchemy import Column, Integer, String, DateTime, Boolean, Text
from sqlalchemy.sql import func
from app.db.base import Base


class Event(Base):
    __tablename__ = "events"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    description = Column(Text, nullable=True)
    image_url = Column(String, nullable=True)
    event_type = Column(String, default="general")  # general, promotion, notice
    
    start_date = Column(DateTime(timezone=True), nullable=True)
    end_date = Column(DateTime(timezone=True), nullable=True)
    
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: surplus-hub-react/packages/core/src/types/event.ts
import { PaginationMeta } from "./material";

export interface Event {
  id: string;
  title: string;
  description: string;
  imageUrl?: string;
  startDate: string;
  endDate: string;
  isActive: boolean;
  createdAt: string;
}

export interface EventsResponse {
  data: Event[];
  meta?: PaginationMeta;
}

export interface EventsQueryParams {
  page?: number;
  limit?: number;
}
